### DISCUSSION TITLE
Monitor and adjust serial/canbus port settings

### DISCUSSION THREAD

**Discussion by MatthewReed303** [cockpit-project/cockpit#17779] [General]
Hi, I'm just wondering if there is a way or a plugin that allows you to monitor serial ports and possibly canbus? The linux kernel has built in API for canbus. Being able to edit the serial port settings such as baud/flow etc would be great addition. Even USB plugin serial devices. They normally show up as ttys0 or ttys1 etc

**Comment by martinpitt** [comment]
There is no such plugin at the moment, see the [applications list](https://cockpit-project.org/applications.html) for the known ones. So feel free to create your own, either the [simple](https://cockpit-project.org/blog/creating-plugins-for-the-cockpit-user-interface.html) or the [comprehensive way](https://cockpit-project.org/blog/cockpit-starter-kit.html).

[cockpit's API](https://cockpit-project.org/guide/latest/api-base1.html) makes it very easy to call programs such as `setserial`. However, we don't currently have an API for udev, i.e. for enumerating, querying, and monitoring device nodes. For that you currently need to call `udevadm`, which is a bit harder to use than libudev. We even [do that on the Storage page](https://github.com/cockpit-project/cockpit/blob/main/pkg/storaged/luksmeta-monitor-hack.py) if you need some inspiration :grin:
